pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -604,6 +625,24 @@ private void doAuthentication(PGStream pgStream, String host, String user, Prope
                 sspiClient.continueSSPI(l_msgLen - 8);
                 break;
 
+              case AUTH_REQ_SASL:
+                if (LOGGER.isLoggable(Level.FINEST)) {
+                  LOGGER.log(Level.FINEST, " <=BE AuthenticationSASL");
+                }
+
+                scramAuthenticator = new ScramAuthenticator(user, password, pgStream);
+                scramAuthenticator.processServerMechanismsAndInit();
+                scramAuthenticator.sendScramClientFirstMessage();
+                break;
+
+              case AUTH_REQ_SASL_CONTINUE:
+                scramAuthenticator.processServerFirstMessage(l_msgLen - 4 - 4);
```

Comment:
Check the message format. It is two ints. I hate this format, but the code is spread with this kind of calculations all the time. I wouldn't do it like this, but I don't want to refactor all this code here, so I just followed the pattern....